blog/content/post/2023-09-12-simplifying-firmware-update.md

```diff
@@ -0,0 +1,111 @@
+between those two.
+
+While software typically resides on easily re-writable, fast, dense media,
+firmware is usually stored separately. On x86, the boot firmware resides in SPI
+flash chips, which is typically mapped by the chipset just below the first 4GB
+of memory, which is a legacy of early x86 processors.
+
+## Status quo
+
+Right now, the recommended method of updating Dasharo is to use flashrom, which
+is a tool running on Linux, writing to the BIOS flash memory by directly
+accessing the SPI flash controller. This has the benefit of removing unnecessary
+complexity from the firmware itself, and giving maximum control to the user. But
+this approach has one big drawback: it's not secure. A malicious actor can
+simply overwrite firmware with malicious, unverified code and we wouldn't be
+able to stop them - and because we don't use Intel Boot Guard or other hardware
+based root of trust, we wouldn't be able to detect such modifications.
+
+How do we protect against this? Dasharo has several security features for this
+specifically:
+
+- Protected Range Registers: The firmware programs which ranges of SPI flash
+  addresses are not allowed to be written to. This is typically set to the
+  initial bootblock, which verifies the subsequent boot stages, and a recovery
+  partition for when verification fails.
+
+- SMM_BWP: This feature only allows code running in System Management Mode to
+  write to the BIOS flash. This code is responsible for EFI variables, which
+  are stored in the BIOS flash. Whenever the OS attempts to enable writes to the
+  SPI flash, an SMI is fired and a handler, installed by the firmware, checks if
+  the write attempt was made in SMM. If it wasn't, then the SMI handler disables
+  writes to BIOS flash and returns control to OS.
+
+- UEFI Secure Boot: This feature verifies the OS's bootloader using a
+  cryptographic signature. The Linux kernel enables Lockdown mode when SB is
+  enabled, which among other things disables direct access to /dev/mem, which
+  prevents access to the SPI flash controller.
+
+![img](/img/whack-a-mole.jpg)
+_Pictured: An accurate summary of SMM_BWP_
+
+All these features prevent a malicious actor from installing malicious firmware,
+but it also prevents the user from updating firmware. Up until this point, we
+have been recommending that users enable these locks for daily usage, but disable
+them whenever they want to update firmware. Recognizing that this can be an
+inconvenient, error prone process, we've added a new feature to make it easier:
+Firmware Update Mode.
+
+## Firmware Update Mode
+
+Firmware Update Mode is an one-time boot mode that disables firmware protections
+for the duration of one boot, allowing the user to boot into a firmware update
+environment (Dasharo Tools Suite or OS of their own choice), update to the
+latest version and reboot, which automatically exits Firmware Update Mode. Here
+is a demo of the new FW update flow on NovaCustom laptops:
```

Comment:
Please add a CTA with links, and information that if someone owns a NovaCustom laptop with a Dasharo, they can experience the simplicity of a new Firmware Update Mode themselves. However, please include a note that this feature is available from the releases between v1.7.0 and v1.8.0+ for NovaCustom NV4x ADL (12th Gen) and NovaCustom NS5x/NS7x ADL (12th Gen), as well as between v1.5.0 and v1.6.0+ for NovaCustom NS5x/NS7x TGL (11th Gen) and NovaCustom NV4x TGL (11th Gen). Additionally, please add links to the release notes for the respective hardware in the description on docs.dasharo. 

Maybe something like (but somehow shorter):

> Firmware Update Mode is now available for NovaCustom laptops with Dasharo firmware. For NovaCustom NV4x ADL (12th Gen) and NovaCustom NS5x/NS7x ADL (12th Gen), you can access it in firmware versions between v1.7.0 and v1.8.0+. If you have NovaCustom NS5x/NS7x TGL (11th Gen) or NovaCustom NV4x TGL (11th Gen), this feature can be found in firmware versions between v1.5.0 and v1.6.0+
> For detailed information on these firmware releases, visit our documentation page [here](https://docs.dasharo.com/unified/novacustom/overview/). Now, you can ensure your NovaCustom laptop is always up-to-date with the latest enhancements and improvements.